runtime/rlog/rlog.go

```diff
@@ -26,7 +26,16 @@ const (
 	levelInfo  logLevel = 2
 	levelWarn  logLevel = 3
 	levelError logLevel = 4
+)
 
+const (
+	// InternalKeyPrefix is the prefix of log field keys that are reserved for
+	// internal use only. For example, logs-based metrics rely on a log field named
+	// 'encore_metric_name' to determine which metric is being emitted.
+	//
+	// If Encore apps add log fields that start with this prefix, we'll prefix them
+	// with 'x_' to make sure they don't interfere with internal log fields.
+	//
 	//publicapigen:drop
 	InternalKeyPrefix = "encore_"
 )
```

Comment:
Nit: No need to use a `const ( ... )` block with a single item. Also, don't use language like "we'll prefix them"
in documentation because it's not clear who "we" refers to.

How about:

```suggestion
// InternalKeyPrefix is the prefix of log field keys that are reserved for
// internal use only. Log fields beginning with this value have an additional
// "x_" prefix prepended to avoid interference with reserved names.
//
//publicapigen:drop
const InternalKeyPrefix = "encore_"
```